### DISCUSSION TITLE
Is there a way to save a map without saving its images in the .db file?

### DISCUSSION THREAD

**Discussion by fernandotorch** [introlab/rtabmap#1305] [Q&A] [answered]
If I understand correctly, rtabmap uses image features for loop closure. However, the original image doesn't need to be kept if the features are stored. In order to preserve the privacy of people around at the time of "recording" the map, is it possible to not save the images and only the features while mapping? Is there otherwise an easy way to delete them afterwards from the db file while still keeping it usable?

**Comment by matlabbe** [comment, marked as answer]
Set parameter `Mem/BinDataKept` to false, then image data won't be saved in the database.

There is indeed no usage of the RGB and depth images (or stereo images) in production unless we want to export high resolution point cloud afterwards. For development, it is useful to keep them for visualization/debugging/reprocessing database with different parameters (e.g., trying other visual features or different parameters to generate the occupancy grid).

**Reply by fernandotorch** [reply]
This is exactly what I was looking for. Thanks a lot for the help!
